### PR TITLE
B2B-712: First pass at the universal ESLint rules

### PR DESCRIPTION
**Pull request by ukmadlz**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        ArrowFunctionExpression: true,
+        FunctionExpression: true
+      }
+    }],
+    'implicit-arrow-linebreak': 0,
+    'operator-linebreak': ['error', 'after', {
+      overrides: {
+        '?': 'before',
+        ':': 'before'
+      }
```
Comment (mariuslazar93) [inline index.js:37]: Can we add
`    "arrow-parens": [
      "error",
      "as-needed",
      { "requireForBlockBody": true }
    ]`

**Review by mariuslazar93 — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "eslint-config-packt-base",
+  "version": "0.0.0",
```
Comment (supachris28) [inline package.json]: version it

File: package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  "description": "Packt's base JS ESLint  config",
+  "main": "index.js",
+  "scripts": {
+    "prelint": "eclint check * rules/* test/*",
+    "lint": "eslint --report-unused-disable-directives .",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (supachris28) [inline package.json]: lint?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "eslint-config-packt-base",
+  "version": "0.0.1",
+  "description": "Packt's base JS ESLint  config",
```
Comment (codebubb) [inline package.json]: Extra space

**Review by mariuslazar93 — DISMISSED** [review]

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "extends": "./index.js",
+    "rules": {
+      // disable requiring trailing commas because it might be nice to revert to
+      // being JSON at some point, and I don't want to make big changes now.
+      "comma-dangle": 0,
```
Comment (supachris28) [inline .eslintrc:6]: should we not be putting these rules in the actual ruleset?

Comment (ukmadlz) [reply .eslintrc:6]: Nope, these are purely to allow it to lint itself when we write custom rules and not have to deal with painful test writing.

**Review by mariuslazar93 — APPROVED** [review]

**Review by codebubb — APPROVED** [review]

**Pull request merged**